string.axi

```diff
@@ -804,4 +804,62 @@ define_function char string_ends_with(char a[], char search[])
 	return right_string(a, length_string(search)) == search;
 }
 
-#end_if
\ No newline at end of file
+
+/**
+ * Remove characters from the end of the string.
+ *
+ * @param	a			the input string
+ * @param	count		the number of characters to remove
+ * @return				the contents of 'a' with the characters removed
+ */
+define_function char[STRING_RETURN_SIZE_LIMIT] strip_chars_right(char a[],
+		integer count)
+{
+	return left_string(a, length_string(a) - count)
+}
+
+/**
+ * Wrapper method for mid_string to bring inline with other programming
+ * languages.
+ *
+ * @param	a			the input string
+ * @param	start		the start location of the substring
+ * @param	count		the number of characters to extract
+ */
+define_function char[STRING_RETURN_SIZE_LIMIT] substr(char a[], integer start,
+		integer count)
+{
+	return mid_string(a, start, count);
+}
+
+/**
+ * Alternative to substr which allows an end location to be specified instead of
+ * a count
+ *
+ * @param	a			the input string
+ * @param	start		the start location of the substring
+ * @param	end			the end location of the substring
+ */
+define_function char[STRING_RETURN_SIZE_LIMIT] substring(char a[],
+		integer start, integer end)
+{
+	return substr(a, start, end-start+1);
+}
+
+define_function CHAR[STRING_RETURN_SIZE_LIMIT] pad_leading_chars(char a[], char pad,

```

Comment:
@THLabs, can you add in some doc for this function (yeah, I'm pedantic, I know).